src/login.ts

```diff
@@ -227,6 +228,12 @@ const states = [
       debug("Typing password");
       await page.keyboard.type(password);
 
+      // federation with on-prem ADFS causes an HTTP basic auth prompt inside a redirect here
+      await page.authenticate({
+        username: String(username),
+        password: String(password),
+      });
```

Comment:
This can be:

```js
await page.authenticate({
  username,
  password,
});
```

If you initialize the password variable as a string: 
```js
let password: string;
```